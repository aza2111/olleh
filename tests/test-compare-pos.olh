//test-compare-pos: compare ints and chars

if (1 < 2):
        if (1 <= 2):
                if (1 != 2):
                        if (2 > 1):
                                if (2 >= 1):
                                        print "1<2; 1<=2; 1!=2; 2>1; 2>=1";
                                end;
                        end;
                end;
        end;
end;

int x = 2;
if (x == 2):
        print "x is equal to 2!";
end;

if ('a' < 'b'):
        if ('a' <= 'b'):
                if ('a' != 'b'):
                        if ('b' > 'a'):
                                if ('b' >= 'a'):
                                        print "'a'<'b'; 'a'<='b'; 'a'!='b';" +
                                                " 'b'>'a'; 'b'>='a'";
                                end;
                        end;
                end;
        end;
end;

char c = 'c';
if (c == 'c'):
        print "c == 'c'!";
end;
