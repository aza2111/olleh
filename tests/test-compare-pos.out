1<2; 1<=2; 1!=2; 2>1; 2>=1
x is equal to 2!
'a' < 'b'
'a' <= 'b'
'a' != 'b'
'a'<'b'; 'a'<='b'; 'a'!='b'; 'b'>'a'; 'b'>='a'
c == 'c'!
